### PR TITLE
Documentation Structure

### PR DESCRIPTION
**Pull request by cloud-j-luna**

### CODE REVIEW

**Comment by anilmurty** [comment]
The high level structure makes sense but I'm not sure about the steps inside "getting started". Some questions:
1. Do we intend to pick one type of deploy option and have the user try that (the easiest one that is best for a new user)?
2. "set up environment" sounds heavy (and development env like) - would make sense for a developer portal (or the developer section of the docs) unless this is the intent here.
3. Choosing a few examples: like dpeloying a static website, deploying a game etc may be a faster ramp

**Comment by cloud-j-luna** [comment]
>     1. Do we intend to pick one type of deploy option and have the user try that (the easiest one that is best for a new user)?
We could include all available deploy options under a "Clients" section. The "Getting Started" section is aimed to allow users to deploy on Akash by simply running a couple commands. The focus on the CLI is important here (although it should rely on a new simplified CLI). The user should have a deployment running on Akash under 3 minutes.
>     2. "set up environment" sounds heavy (and development env like) - would make sense for a developer portal (or the developer section of the docs) unless this is the intent here.
"Environment Setup" is what I can come up with. I will update the docs but mostly this sections sets the environment variables for the user (`.bashrc`, `.zshrc`, ...), creating the certificate, etc...
>     3. Choosing a few examples: like dpeloying a static website, deploying a game etc may be a faster ramp
These examples could be other section called "Examples"

Feel free to commit to this branch as well.

File: sig-documentation/documentation-structure.md [hunk 1]
```diff
@@ -57,16 +56,50 @@ The following table is a tabulation for the requirements.
...
+    * Overview
+    * Kubernetes & Containers
+    * Akash Node
+    * Akash Provider
+5. Guides
+    * Deploy using the CLI
+    * Deploying using Cloudmos
+    * Deploying using Terraform
+6. Developers
+    * _This is the scope of the client libraries documentation_
```
Comment (anilmurty) [inline sig-documentation/documentation-structure.md:88]: Was this a placeholder meant to be replaced with something else? Or are you saying that this is out of scope for the doc site?

File: sig-documentation/documentation-structure.md [hunk 2]
```diff
@@ -57,16 +56,50 @@ The following table is a tabulation for the requirements.
...
+7. Providers
+    * Setting up a Provider with Helm
+    * Setting up a Provider with Praetor
+    * Enabling monitoring
+    * Provider Auditing
+    * Persistent Storage
```
Comment (anilmurty) [inline sig-documentation/documentation-structure.md:94]: "Enabling" Persistent Storage?

File: sig-documentation/documentation-structure.md [hunk 3]
```diff
@@ -57,16 +56,50 @@ The following table is a tabulation for the requirements.
...
+    * Enabling IP Leasing
 
 ## Usability
 
 ### Learnability
 
-...
\ No newline at end of file
+The first chapter (Getting Started) should allow users to have a running deployment in less than 5 minutes.
+It should also permanently setup the user's environment so that the development environment still works on a full restart.
```
Comment (anilmurty) [inline sig-documentation/documentation-structure.md:102]: I think the goal of having a "permanent setup" ADN getting them running as fast as possible (<5min) may be at odds at each other. Note that the assumption (stated above in this doc) is that the first time user is not familiar with crypto even

File: sig-documentation/documentation-structure.md (hunk 1 above)
Comment (cloud-j-luna) [reply sig-documentation/documentation-structure.md:88]: It's out of this scope as this will be addressed in the Client Libraries' documentation. It will depend on the interface that will be provided.

File: sig-documentation/documentation-structure.md [hunk 4]
```diff
@@ -57,16 +56,50 @@ The following table is a tabulation for the requirements.
...
 
 The proposed structure is as follows:
-
 1. Getting Started
-* Setting up the Environment
-* First Deployment
+    * Setting up the Environment
+    * Your First Deployment
 2. Concepts
+    * Akash Network
```
[the remark this answers is not in the document]
Comment (cloud-j-luna) [reply sig-documentation/documentation-structure.md:66]: I think that's in scope of the "Bids & Leases" section, do you agree?

File: sig-documentation/documentation-structure.md (hunk 3 above)
Comment (cloud-j-luna) [reply sig-documentation/documentation-structure.md:102]: For instance if the documentation simply addresses changing environment variables the environment will be lost on restart. By writing `export`s on the `.bashrc`, `.zshrc`, ... files will set those variables every time a new terminal session is started and thus the user will have the environment available "permanently". They could shut off their computer and the next day still have a working environment when they open a terminal.

File: sig-documentation/documentation-structure.md (hunk 1 above)
Comment (anilmurty) [reply sig-documentation/documentation-structure.md:88]: So the "6. Developers" section is out of scope? If so, should it just be removed from this doc?

File: sig-documentation/documentation-structure.md (hunk 3 above)
Comment (anilmurty) [reply sig-documentation/documentation-structure.md:102]: not questioning the technical details, just wondering if it is more important to get people runnning quickly or to get their set up such that they are set up for being able to do more complicated tasks later. I think it may be hard to achieve both objectives

File: sig-documentation/documentation-structure.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (anilmurty) [reply sig-documentation/documentation-structure.md:66]: agreed.

File: sig-documentation/documentation-structure.md (hunk 3 above)
Comment (cloud-j-luna) [reply sig-documentation/documentation-structure.md:102]: Good point, although I believe we can/should strive for achieving both. These first sections are very important and we must provider the best experience we can through, docs/CLI/network experience.

File: sig-documentation/documentation-structure.md (hunk 1 above)
Comment (cloud-j-luna) [reply sig-documentation/documentation-structure.md:88]: The definition of the section is out of scope but its presence in the overall structure of the documentation isn't.

**Review by chainzero — COMMENTED** [review]
Concentrating review on the proposed structure - I think the structure is excellent.  Begin with a better Getting Started section than is available today and logical sectioning as the user continues into Akash Concepts, Features, Architecture, etc.  And then into how to/guides.

I would be excited - and know this is the intent - to now bring this to our users strategically for thoughts.  I.e. possibly presenting to insiders for comments and other users we are close with for thoughts on this documentation structuring.

Believe current version is sufficient to begin review with users and would assume there will be considerable change/additions coming out of those sessions.

File: sig-documentation/documentation-structure.md (hunk 1 above)
Comment (anilmurty) [reply sig-documentation/documentation-structure.md:88]: OK, so I would suggest rephrasing to "Developer documentation is covered in Client Libraries Documentation" (and we can link to the client lib docs, in the actual implementation of this)

Comment (cloud-j-luna) [reply sig-documentation/documentation-structure.md:88]: This section will include all the documentation to use the client libraries (in a guide like structure)

**Review by anilmurty — APPROVED** [review]
